### PR TITLE
Specify and implement DCRCP

### PR DESCRIPTION
**Pull request by adam-dej**
This PR specified and implements DeadCom Controller-Reader Protocol.

It is currently work-in-progress. Missing stuff:

- [x] Integration testing with ChibiOS (on CortexM0 MCUs, there may be some issues with unaligned access)
- [x] Maybe cn-cbor can be slightly modified to be usable with ChibiOS Memory Pools instead of Heap Allocator?
- [x] in `dcrcpDecode` either zero CRPM struct in this library, or mention in documentation that the struct must be memset to zero

### CODE REVIEW

File: dcrcp/lib/src/dcrcp.c [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+            if (!cn_cbor_mapput_int(crpm, DCRCP_CRPM_SYS_QUERY_RESPONSE, sqr, ctx, &err)) {
+                CHECK_ERROR_ENCODER(NULL, err);
+            }
+            break;
+        }
+        case DCRCP_CRPM_ACTIVATE_AUTH_METHOD: {
+            cn_cbor *ams = cn_cbor_array_create(ctx, &err);
+            CHECK_ERROR_ENCODER(ams, err);
+
+            size_t i;
```
Comment (lubo) [inline dcrcp/lib/src/dcrcp.c]: How about moving this definition into the for cycle below?

File: dcrcp/lib/src/dcrcp.c [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+            if (!cn_cbor_mapput_int(crpm, DCRCP_CRPM_UI_UPDATE, ui_state, ctx, &err)) {
+                CHECK_ERROR_ENCODER(NULL, err);
+            }
+            break;
+        }
+        case DCRCP_CRPM_AM0_PICC_UID_OBTAINED: {
+            cn_cbor *uids = cn_cbor_array_create(ctx, &err);
+            CHECK_ERROR_ENCODER(uids, err);
+
+            size_t i;
```
Comment (lubo) [inline dcrcp/lib/src/dcrcp.c]: The same as before.

File: dcrcp/lib/src/dcrcp.c (hunk 1 above)
Comment (adam-dej) [reply dcrcp/lib/src/dcrcp.c]: Fixed

File: dcrcp/lib/src/dcrcp.c (hunk 2 above)
Comment (adam-dej) [reply dcrcp/lib/src/dcrcp.c]: Fixed

**Comment by adam-dej** [comment]
I've implemented required fixes and rebased this branch. Can you please take a look again @lubo ?

**Review by lubo — APPROVED** [review]
Sure. LGTM now.

**Pull request merged**